pkg/apis/apps/types.go

```diff
@@ -97,6 +97,40 @@ type RollingUpdateStatefulSetStrategy struct {
 	Partition int32
 }
 
+// PersistentVolumeClaimDeletePolicyType is a string enumeration of the policies
+// for deletion of VolumeClaimTemplates when the controlling StatefulSet is
+// deleted or scaled down.
+type PersistentVolumeClaimDeletePolicyType string
+
+const (
+	// RetainPersistentVolumeClaimDeletePolicyType is the default
+	// PersistentVolumeClaimDeletePolicy and specifies that
+	// PersistentVolumeClaims associated with StatefulSet VolumeClaimTemplates
+	// will not be deleted.
+	RetainPersistentVolumeClaimDeletePolicyType PersistentVolumeClaimDeletePolicyType = "Retain"
+	// DeletePersistentVolumeClaimDeletePolicyType specifies that
+	// PersistentVolumeClaims associated with StatefulSet VolumeClaimTemplates
+	// will be deleted in the scenario specified in
+	// StatefulSetPersistentVolumeClaimPolicy.
+	DeletePersistentVolumeClaimDeletePolicyType PersistentVolumeClaimDeletePolicyType = "Delete"
+)
+
+// StatefulSetPersistentVolumeClaimPolicy describes the policy used for PVCs
+// created from the StatefulSet VolumeClaimTemplates.
+type StatefulSetPersistentVolumeClaimPolicy struct {
+	// OnSetDeletion specifies what happens to PVCs created from StatefulSetwhen
```

Comment:
A space is missing in `StatefulSetwhen`. But it looks like you forgot to clean up some words. Same for `OnScaledown`